.github/workflows/build.yml

```diff
@@ -11,15 +16,86 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        node_version: [18.x, 20.x]
+        node_version: [22.x, 20.x]
         os: [ubuntu-latest, macos-latest, windows-latest]
     steps:
       - uses: actions/checkout@v4
-      - name: Use Node.js ${{ matrix.node-version }}
+        with:
+          fetch-depth: 0  # Fetch all history for checking version changes
+      
+      - name: Use Node.js ${{ matrix.node_version }}
         uses: actions/setup-node@v4
         with:
           node-version: ${{ matrix.node_version }}
+      
       - run: npm install
+      
       - run: npm test
         env:
           CI: true
+  publish:
+    name: Publish to NPM
+    needs: build  # Wait for all build matrix jobs to complete successfully
+    # Only run when:
+    # 1. The code is on the main branch
+    # 2. It's either a direct push or a merged PR
+    # 3. All 6 matrix checks have passed (enforced by 'needs: build')
+    if: |
+      (github.ref == 'refs/heads/main') && 
+      (github.event_name == 'push' || 
+       (github.event_name == 'pull_request' && github.event.pull_request.merged == true))
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Fetch all history for checking version changes
+      - name: Check for version change
```

Comment:
I would suggest to create a separate `release.yml` GH Action configuration that gets triggered when tagged with `*.*.*` instead of programmatically checking if version is changed.

Reply:
Hi @sivaprasadreddy, 

Addressed the same. Can you please review and approve the same.